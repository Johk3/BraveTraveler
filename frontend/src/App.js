import React from 'react';
import logo from './logo.svg';
import CanvasJSReact from './canvasjs.react'
import './App.css';

var CanvasJSChart = CanvasJSReact.CanvasJSChart;


class App extends React.Component {
  constructor() {
    super();
    this.generateDataPoints = this.generateDataPoints.bind(this);
  }

  generateDataPoints(noOfDps) {
    var xVal = 1, yVal = 100;
    var dps = [];
    for(var i = 0; i < noOfDps; i++) {
      yVal = yVal +  Math.round(5 + Math.random() *(-5-5));
      dps.push({x: xVal,y: yVal});	
      xVal++;
    }
    return dps;
  }

  render() {
    const options = {
      theme: "dark1", // "light1", "dark1", "dark2"
      animationEnabled: true,
      zoomEnabled: true,
      title: {
        text: "Audio Interception",
      },
      axisY: {
        includeZero: false,
        title: "Decibels",
      },
      axisX: {
        title: "Iterations",
      },
      data: [{
        type: "area",
        dataPoints: this.generateDataPoints(500),
      }]
    }

    return(
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <h1>
          BraveTraveler - Monitoring
        </h1>
        <div style={{columns: "2"}}>
          <CanvasJSChart options={options}/>
          <CanvasJSChart options={options}/>
          <CanvasJSChart options={options}/>
          <CanvasJSChart options={options}/>
        </div>
      </header>
    </div>
    )
  }
}

export default App;
