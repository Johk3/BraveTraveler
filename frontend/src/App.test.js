import React from 'react';
// import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import App from './App';

describe('MyComponent', () => {
  it('should render correctly in "debug" mode', () => {    const component = shallow(<App debug />);
  
    expect(component).toMatchSnapshot();
  });
  it('should render correctly with no props', () => {
    const component = shallow(<App/>);

    expect(component.find("h1").text()).toContain("BraveTraveler - Monitoring")
  });
});
// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
