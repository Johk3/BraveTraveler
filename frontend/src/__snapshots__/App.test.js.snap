// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1>
      BraveTraveler - Monitoring
    </h1>
    <Copyright />
  </header>
</div>
`;
